PCI-1428 Disable manual trigger on the merge upstream job

aliculPix4D opened the pull request:
- it should not be possible to run merge upstream job from a feature pipeline

Pix4D-Janus commented:
Your pipeline is set.
Check it here:https://builder.ci.pix4d.com/teams/developers/pipelines/dependabot-PCI-1428-disable-manual-trigger

manuel-pix4d commented on ci/pipelines/dependabot-template.yml:
```diff
@@ -98,6 +98,7 @@ jobs:
       file: dependabot-core.git/ci/tasks/test-dependabot-task.yml
 
   - name: merge-upstream
+    disable_manual_trigger: ((branch-is-master))
```
So shouldn't be the other way around ? Because `disable_manual_trigger` is set to `true` when we are on master. So on feature, `disable_manual_trigger` is set to `false`, so it's enable right ? I'm lost with all the negative inclusion :D

aliculPix4D replied:
Yes, you are right. I will change it. Thanks  :)

manuel-pix4d replied:
Then rename the parameter because you describe here the opposite: true if feature and false if master.

aliculPix4D replied:
`disable_manual_trigger: true` for all pipelines, feature and master. No need to rename anything.

aliculPix4D commented on ci/pipelines/dependabot-template.yml:
```diff
@@ -90,14 +90,15 @@ jobs:
     plan:
     - get: dependabot-core.git
-      trigger: ((trigger))
+      trigger: true
```
Also, we should trigger test job on each commit, with no need for manual operations.

manuel-pix4d reviewed (commented):
Are you sure you want to allow manual triggering, even only for master branch ? Shouldn't it be only automated operations ?
As for the merge-upstream task, I would put it *after* the testing task. Is there a reason why they are in parallel ?

aliculPix4D commented:
> Are you sure you want to allow manual triggering, even only for master branch ? Shouldn't it be only automated operations ?
I have disabled manual triggering, completely so now it can be triggered only by changes upstream. 

> As for the merge-upstream task, I would put it _after_ the testing task. Is there a reason why they are in parallel ?
They are not parallel. As I see it, this are two independent jobs in a single pipeline. Only tests are important for the feature pipeline, hence I wanted to block that this job can be run from feature pipelines. This is done by setting trigger false on all resources for the merge-upstream job and setting `disable_manual_trigger: true`

manuel-pix4d commented:
> Only tests are important for the feature pipeline

OK so you assume they are run in feature pipeline before any merge. To me it's a safety protection to run the test every time before merging anything. But I'm OK let it be like this !

aliculPix4D commented:
@manuel-pix4d @aledeganopix4d @odormond I have updated the PR since you reviewed it. Instead of seting trigger to true, I have added to the feature settings file
```
pr_bot_triggered_jobs:
    - test-dependabot
```

manuel-pix4d commented:
Nice ! Didn't know about this `pr_bot_triggered_jobs` parameters ! A reason more to attend to the automation tools presentation :)
